File: backend/src/models/pipeline/PipelineCsvInfo.ts
import { BuildInfo } from "./BuildInfo";
import { DeployInfo } from "./DeployTimes";
import { CommitInfo } from "../codebase/CommitInfo";
import { LeadTimeInfo } from "../codebase/LeadTime";

export class PipelineCsvInfo {
  pipeLineName: string;
  stepName: string;
  buildInfo?: BuildInfo;
  deployInfo?: DeployInfo;
  commitInfo?: CommitInfo;
  leadTimeInfo?: LeadTimeInfo;

  constructor(
    pipeLineName: string,
    stepName: string,
    buildInfo: BuildInfo,
    deployInfo: DeployInfo,
    commitInfo: CommitInfo,
    leadTimeInfo: LeadTimeInfo
  ) {
    this.pipeLineName = pipeLineName;
    this.stepName = stepName;
    this.buildInfo = buildInfo;
    this.deployInfo = deployInfo;
    this.commitInfo = commitInfo;
    this.leadTimeInfo = leadTimeInfo;
  }
}

File: backend/src/main/java/heartbeat/controller/report/dto/response/PipelineCSVInfo.java
package heartbeat.controller.report.dto.response;

import heartbeat.client.dto.codebase.github.CommitInfo;
import heartbeat.client.dto.pipeline.buildkite.BuildKiteBuildInfo;
import heartbeat.client.dto.pipeline.buildkite.DeployInfo;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PipelineCSVInfo {

	private String pipeLineName;

	private String stepName;

	private Boolean valid;

	private BuildKiteBuildInfo buildInfo;

	private DeployInfo deployInfo;

	private CommitInfo commitInfo;

	private LeadTimeInfo leadTimeInfo;

}
